app/api/groups/create_group.robustness.feature

```diff
@@ -0,0 +1,185 @@
+      | 21      | 10     | 2019-07-16T21:28:47Z  | null                     | null                    |
+      | 21      | 11     | null                  | 2019-07-16T21:28:47Z     | null                    |
+      | 21      | 12     | null                  | null                     | 2019-07-16T21:28:47Z    |
+
+  Scenario: No name
+    Given I am the user with ID "1"
+    When I send a POST request to "/groups" with the following body:
+    """
+    {"type": "Class"}
+    """
+    Then the response code should be 400
+    And the response body should be, in JSON:
+    """
+    {
+      "success": false,
+      "message": "Bad Request",
+      "error_text": "Invalid input data",
+      "errors": {"name": ["missing field"]}
+    }
+    """
+    And the table "groups" should stay unchanged
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Empty name
+    Given I am the user with ID "1"
+    When I send a POST request to "/groups" with the following body:
+    """
+    {"name": "", "type": "Class"}
+    """
+    Then the response code should be 400
+    And the response body should be, in JSON:
+    """
+    {
+      "success": false,
+      "message": "Bad Request",
+      "error_text": "Invalid input data",
+      "errors": {"name": ["name must be at least 1 character in length"]}
+    }
+    """
+    And the table "groups" should stay unchanged
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: No type
+    Given I am the user with ID "1"
+    When I send a POST request to "/groups" with the following body:
+    """
+    {"name": "some name"}
+    """
+    Then the response code should be 400
+    And the response body should be, in JSON:
+    """
+    {
+      "success": false,
+      "message": "Bad Request",
+      "error_text": "Invalid input data",
+      "errors": {"type": ["missing field"]}
+    }
+    """
+    And the table "groups" should stay unchanged
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario Outline: Empty or wrong type
+    Given I am the user with ID "1"
+    When I send a POST request to "/groups" with the following body:
+    """
+    {"name": "some name", "type": "<type>"}
+    """
+    Then the response code should be 400
+    And the response body should be, in JSON:
+    """
+    {
+      "success": false,
+      "message": "Bad Request",
+      "error_text": "Invalid input data",
+      "errors": {"type": ["type must be one of [Class Team Club Friends Other]"]}
+    }
+    """
+    And the table "groups" should stay unchanged
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+  Examples:
+    | type    |
+    |         |
+    | Unknown |
```

Comment:
Also, some types are forbidden (for instance "RootSelf"). Only types from [ 'Class','Team','Club','Friends','Other'] are allowed. I don't think this is tested.

Reply:
fixed